### PR TITLE
Removal of unused code and full coverage in physics units

### PR DESCRIPTION
**Pull request by jirikuncar**
cherry-picked from #12970 

--
cc @smichr @asmeurer @Upabjojr

### CODE REVIEW

**Comment by jirikuncar** [comment]
Build timeouted on travis: https://travis-ci.org/sympy/sympy/jobs/255538428

**Comment by jirikuncar** [comment]
@asmeurer here is another example why it is important to regularly merge maintenance branch back to master. The commit 031e0df is a duplicate of #13009 but without it the tests will fail.

**Comment by jirikuncar** [comment]
Build timeouted on travis: https://travis-ci.org/sympy/sympy/jobs/255938088

File: sympy/physics/units/dimensions.py [hunk 1]
```diff
@@ -206,8 +207,6 @@ def _get_dimensional_dependencies_for_name(name):
             return {k: v for (k, v) in ret.items() if v != 0}
 
         if name.is_Pow:
-            if name.exp == 0:
-                return {}
```
Comment (Upabjojr) [inline sympy/physics/units/dimensions.py]: why this removal? You could have `Pow(x, 0, evaluate=False)`

File: sympy/physics/units/quantities.py [hunk 2]
```diff
@@ -83,10 +83,7 @@ def scale_factor(self):
         return self._scale_factor
 
     def _eval_is_positive(self):
-       return self.scale_factor.is_positive
-
-    def _eval_is_constant(self):
```
Comment (Upabjojr) [inline sympy/physics/units/quantities.py]: Why this removal?

File: sympy/physics/units/tests/test_dimensions.py [hunk 3]
```diff
@@ -73,3 +84,6 @@ def test_mul_div_exp():
 
     assert length != 1
     assert length / length != 1
+
+    length_0 = length ** 0
+    assert length_0.get_dimensional_dependencies() == {}
```
Comment (Upabjojr) [inline sympy/physics/units/tests/test_dimensions.py]: `Pow(length, 0, evaluate=False)`

**Comment by Upabjojr** [comment]
It looks OK apart from some minor doubts.

File: sympy/physics/units/dimensions.py (hunk 1 above)
Comment (jirikuncar) [inline sympy/physics/units/dimensions.py]: There might be expressions in dimensions, which I think should be exposed to the multiplication by zero.

File: sympy/physics/units/tests/test_dimensions.py (hunk 3 above)
Comment (jirikuncar) [inline sympy/physics/units/tests/test_dimensions.py]: Can you show me some real life use-case? It looks like you would like to test the internals of `_get_dimensional_dependencies_for_name`.

Comment (Upabjojr) [inline sympy/physics/units/tests/test_dimensions.py]: I noticed that you have removed the `expr.exp == 0` case. The first thought I had was that case, so I just wrote that. But I think that's not important at all.

File: sympy/physics/units/quantities.py (hunk 2 above)
Comment (Upabjojr) [inline sympy/physics/units/quantities.py]: You're right, we don't need this.

**Comment by Upabjojr** [comment]
OK. I think we can merge this.

**Pull request merged**